### ISSUE TITLE
Proxy is very slow -1s compared to 5ms without it.

### ISSUE THREAD

**Issue by kmolerov** [webpack/webpack-dev-server#1225]
* Operating System: windows 8
* Node Version: 8.9.1
* NPM Version: 5.5.1
* webpack Version: 3.10.0 
* webpack-dev-server Version: 2.9.7

- [x] This is a **bug**
- [ ] This is a **feature** request
- [ ] This is a **modification** request

### Code

```js
  // webpack.config.js
"devServer": {
	  proxy: [{
      "path": "/api/*",
      "target": "http://localhost:10300",
	  "secure": false,
      "logLevel": "silent",
      "ws": false
    }]
}
```

### Expected Behavior
proxy to rest api should be very fast  (should add no overhead)
### Actual Behavior
proxy to rest api is very slow (1s is spend waiting for response)
### For Bugs; How can we reproduce the behavior?
Just call any rest api trough the proxy of dev servere and compare it with call directly to the rest api.

### For Features; What is the motivation and/or use-case for the feature?
Single page application that should load data from rest service. Proxy is used during development, so we are not configuring dev urls in clinet and CORS in server.

Note: problem is not in the http-proxy-middleware, that dev server uses. I have tried proxy with http-proxy-middleware and it works instantly without any added overhead.

**Comment by shellscape** [comment]
@kmolerov thanks for checking in. a few things;

- we're not overly concerned about speed given that this is a dev-environment-only server, and some delays are acceptable.
- proxy issues should be taken up with the module which handles the proxies, [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware). webpack-dev-server uses that module directly.

we've not experience the same kind of dramatic slowdown, so you probably have some kind of a unique situation causing that with http-proxy-middleware. please head to that project to report the issue.

**Comment by kmolerov** [comment]
dev-environment-only server should be the fastest of all, as i am calling it hundreds of times a day and I am loosing time and concentration if I have to wait for slow requests.

I don't think the problem is with http-proxy-middleware, as I have tried it stand-alone without web-pack and it works great. 
I have found that during the initial load after loading some static file (for example http://localhost:4200/main.bundle.js.map) first call to the proxy rest api is instant (3-6ms), after that all calls are taking 1 s. If I load some static file again I am observing the same pattern.
